kernel/net/net_dhcp.h

```diff
@@ -147,7 +147,7 @@ int net_dhcp_init(void);
 
 void net_dhcp_shutdown(void);
 
-int net_dhcp_request(void);
+int net_dhcp_request(uint32* required_address);
```

Comment:
I'm curious why the parameter here is a pointer. Seeing how it is used, I think it would be better to just pass the normal packed form of an IPv4 address here. Especially since passing 0 would make the current behavior free, without needing the `?` that was added into the function.

Reply:
Fixed.